src/account/items.ts

```diff
@@ -184,4 +204,28 @@ export default class Items<P extends Item, I extends Item> {
     }
     return q;
   }
+
+  _getId(id: string, partition: PartitionValue) {
+    return [partition || id, id].join(":");
+  }
+
+  _getPartition(data: { [x: string]: any }): PartitionValue {
+    return getValue(this._getPartitionKeyPath(), data);
+  }
+
+  _getPartitionKeyPath(): string[] {
+    try {
+      const [firstPath] = this._partitionKeyPath;
+      if (!firstPath) {
+        return ["id"]; // Default/fallback to `id` as partition key for now
+      }
+      const path = firstPath.slice(1).split("/");
+      if (path.length === 1 && path[0] === "_partitionKey") {
+        return ["id"];
+      }
+      return path;
+    } catch (error) {
+      return ["id"];
+    }
```

Comment:
When does this can throw error? It'd be great if we can handle it without try-catch. If it's not possible, I think we should narrow the scope of try-catch to not suppress unexpected errors.

Reply:
@nkzawa ill remove this - was left over from testing